src/hash.rs

```diff
@@ -20,13 +21,15 @@ pub fn permute(h: Hash, s: State) -> State {
   match h {
     Hash::Poseidon2 => poseidon2::permutation::permute(s),
     Hash::Griffin   =>   griffin::permutation::permute(s),
+    Hash::Skyscraper   =>   skyscraper::permutation::permute_state(s),
```

Comment:
We probably shouldn't do this, as it's totally unsafe

Instead somehow make the State type, or maybe just its width to depend on the Hash enum somehow (probably a trait or something?)

Until we figure that out maybe just comment this out